forc-pkg/src/manifest/mod.rs

```diff
@@ -1148,11 +1186,500 @@ pub fn find_dir_within(dir: &Path, pkg_name: &str) -> Option<PathBuf> {
     find_within(dir, pkg_name).and_then(|path| path.parent().map(Path::to_path_buf))
 }
 
+/// Helper function to add source (.sw) files to the hash
+fn add_source_files_to_hash(hasher: &mut sha2::Sha256, base_path: &Path) -> Result<()> {
+    use std::collections::BTreeMap;
+    use walkdir::WalkDir;
+
+    // Using BTreeMap for deterministic ordering by relative path
+    let mut source_files = BTreeMap::new();
+
+    // Find src directory
+    let src_path = base_path.join("src");
+    if src_path.exists() {
+        // Collect all .sw files under src/
+        for entry in WalkDir::new(&src_path) {
+            let entry = entry?;
+            let path = entry.path();
+
+            if path.is_file() && path.extension() == Some(std::ffi::OsStr::new("sw")) {
+                // Calculate relative path from base_path for consistent naming
+                let rel_path = path
+                    .strip_prefix(base_path)
+                    .unwrap_or(path)
+                    .to_string_lossy()
+                    .to_string();
+
+                println!("adding {:?}", rel_path);
```

Comment:
do we need this for prod?
If so; should we use the tracing libs print function(s)?